defold-spine/commonsrc/vertices.cpp

```diff
@@ -196,74 +281,84 @@ uint32_t GenerateVertexData(dmArray<SpineVertex>& vertex_buffer, const spSkeleto
-
-            float page_index = 0;
 
             // Computed the world vertices positions for the 4 vertices that make up
             // the rectangular region attachment. This assumes the world transform of the
             // bone to which the slot (and hence attachment) is attached has been calculated
             // before rendering via spSkeleton_updateWorldTransform
-            spRegionAttachment_computeWorldVertices(regionAttachment, slot, scratch.Begin(), 0, 2);
-
-            // Create 2 triangles, with 3 vertices each from the region's
-            // world vertex positions and its UV coordinates (in the range [0-1]).
-            addVertex(&vertex_buffer[vindex++], scratch[0], scratch[1], uvs[0], uvs[1], colorR, colorG, colorB, colorA, page_index);
-            addVertex(&vertex_buffer[vindex++], scratch[2], scratch[3], uvs[2], uvs[3], colorR, colorG, colorB, colorA, page_index);
-            addVertex(&vertex_buffer[vindex++], scratch[4], scratch[5], uvs[4], uvs[5], colorR, colorG, colorB, colorA, page_index);
-
-            addVertex(&vertex_buffer[vindex++], scratch[4], scratch[5], uvs[4], uvs[5], colorR, colorG, colorB, colorA, page_index);
-            addVertex(&vertex_buffer[vindex++], scratch[6], scratch[7], uvs[6], uvs[7], colorR, colorG, colorB, colorA, page_index);
-            addVertex(&vertex_buffer[vindex++], scratch[0], scratch[1], uvs[0], uvs[1], colorR, colorG, colorB, colorA, page_index);
+            spRegionAttachment_computeWorldVertices(regionAttachment, slot, scratch_vertex_floats.Begin(), 0, 2);
+
+            vertex_count  = 4;
+            uvs           = regionAttachment->uvs;
+            indices       = (uint16_t*) QUAD_INDICES;
+            indices_count = 6;
+            color         = &regionAttachment->color;
+            vertices      = scratch_vertex_floats.Begin();
         }
         else if (attachment->type == SP_ATTACHMENT_MESH)
         {
             // Cast to an spMeshAttachment so we can get the rendererObject
             // and compute the world vertices
-            spMeshAttachment* mesh = (spMeshAttachment*)attachment;
+            spMeshAttachment* mesh = (spMeshAttachment*) attachment;
 
             int num_world_vertices = SUPER(mesh)->worldVerticesLength / 2;
+            spVertexAttachment_computeWorldVertices(SUPER(mesh), slot, 0, mesh->super.worldVerticesLength, scratch_vertex_floats.Begin(), 0, 2);
+
+            vertex_count  = num_world_vertices;
+            uvs           = mesh->uvs;
+            indices       = mesh->triangles;
+            indices_count = mesh->trianglesCount;
+            color         = &mesh->color;
+            vertices      = scratch_vertex_floats.Begin();
+        }
+        else if (attachment->type == SP_ATTACHMENT_CLIPPING)
+        {
+            // Clipper setup is very similar to this:
+            // https://github.com/EsotericSoftware/spine-runtimes/blob/4.0/spine-sfml/c/src/spine/spine-sfml.cpp#L293
```

Comment:
I was directed here from the main manual and have structured all the clipping calls according to how they do it, so I expect it to be correct :)